src/grid.cu

```diff
@@ -287,4 +305,203 @@ void Grid::write_grid(std::filesystem::path fname) const {
             fout.write((char*) &vol, sizeof(double));
         }
     }
-}
\ No newline at end of file
+}
+
+// Sub-grid functions
+
+Grid GridManager::add_subgrid(double R_in, double R_out) {
+
+    int sg_idx = in_idx.size();
+
+    if (R_in > R_out) {
+        throw std::runtime_error("Subgrid R_in greater than R_out; check values!");
+    }
+
+    for (int i=0; i<g.NR+2*g.Nghost+1; i++) {
+        if (g.Re(i) > R_in) {
+            
+            if (i <= g.Nghost) {
+                in_idx.push_back(0);
+                break;
+            }
+            else if ((g.Re(i)-R_in) < (R_in-g.Re(i-1))) {
+                in_idx.push_back(i - g.Nghost);
+                break;
+            }
+            else {
+                in_idx.push_back(i-1 - g.Nghost);
+                break;
```

Comment:
One last thing - can you explain this logic to me?

Reply:
It's to find the closest cell edge to R_in between two cell edges, Re(i) and Re(i-1), then set the in_idx to be Nghost cells further back in the grid so that the first active edge is the closest edge to R_in.

Reply:
Ok, that makes sense. I'll add a comment or two the code and then accept this.